### PR TITLE
Handle OIDC authz request with "response_type" missing and "response_mode=form_post"

### LINKED ISSUE #10144: Handle OIDC authz request with "response_type" missing and "response_mode=form_post"
### Describe the bug

This bug causes that OIDC Form-Post profile is currently failing.

**OIDC Test:** oidcc-response-type-missing

**OIDC Test Description:** This test sends an authorization request that is using “response_mode=form_post”, but is missing the response_type parameter. The authorization server must either redirect back with an 'unsupported_response_type' or 'invalid_request' error, or must display an error saying the response type is missing, a screenshot of which should be uploaded.

Keycloak currently redirects to the client with HTTP GET request containing error. This is considered as an issue as Keycloak should either redirect with HTTP POST or display error page.

**The exact failure from OIDC testsuite:**
FAILURE OAuth2-FP-2 Verify authorization endpoint response: CheckCallbackHttpMethodIsPost: The HTTP method used at redirect_uri is not 'POST'

### Version

17.0.0-SNAPSHOT

### Expected behavior

OIDC Test oidcc-response-type-missing passing. Keycloak should either show error page or send response via form_post POST error response (not GET error response as it is today)

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by tnorimat**
Closes #10144

### CODE REVIEW

**Review by mposolda — CHANGES_REQUESTED** [review]
@tnorimat Thanks Takashi!

The changes looks good to me, I just wonder that it can be useful to have also automated test for this in the testsuite? For example `AuthorizationCodeTest` has some tests for "form post" response mode, so maybe new test worth to be added here?

**Comment by tnorimat** [comment]
@mposolda Yes, I will add an automated test.

**Comment by tnorimat** [comment]
@mposolda I've added arquillian integration tests onto `AuthorizationCodeTest`. Could you check them?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/AuthorizationCodeTest.java [hunk 1]
```diff
@@ -158,6 +158,46 @@ public void authorizationRequestInvalidResponseType() throws IOException {
...
+    @Test
+    public void authorizationRequestFormPostResponseModeInvalidResponseType() throws IOException {
+        oauth.responseMode(OIDCResponseMode.FORM_POST.value());
+        oauth.responseType("tokenn");
+        oauth.stateParamHardcoded("OpenIdConnect.AuthenticationProperties=2302984sdlk");
+        UriBuilder b = UriBuilder.fromUri(oauth.getLoginFormUrl());
+        driver.navigate().to(b.build().toURL());
+
+        String sources = driver.getPageSource();
+        System.out.println(sources);
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/AuthorizationCodeTest.java:170]: Minor: Is it possible to remove `System.out.println` calls from this place? It increases log size and it is hard to filter these from the results.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/AuthorizationCodeTest.java [hunk 2]
```diff
@@ -158,6 +158,46 @@ public void authorizationRequestInvalidResponseType() throws IOException {
...
+    @Test
+    public void authorizationRequestFormPostResponseModeWithoutResponseType() throws IOException {
+        oauth.responseMode(OIDCResponseMode.FORM_POST.value());
+        oauth.responseType(null);
+        oauth.stateParamHardcoded("OpenIdConnect.AuthenticationProperties=2302984sdlk");
+        UriBuilder b = UriBuilder.fromUri(oauth.getLoginFormUrl());
+        driver.navigate().to(b.build().toURL());
+
+        String sources = driver.getPageSource();
+        System.out.println(sources);
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/AuthorizationCodeTest.java:189]: Same as above

**Review by mposolda — CHANGES_REQUESTED** [review]
@tnorimat Thanks for adding tests! Added 2 minor comments regarding to remove `System.out.println` calls.

**Comment by tnorimat** [comment]
@mposolda I've fixed the point you've commented.

**Review by mposolda — APPROVED** [review]
@tnorimat Thanks! I hope to merge this once tests pass.

**Pull request merged**